### PR TITLE
feature/270 creacion nuevo challengeStatus en Solution con valor INPROGRESS

### PR DESCRIPTION
**Pull request by Almami679**
Añadido en el enum de challengeStatus el valor de INPROGRESS

### CODE REVIEW

**Review by AlexandraBonetCanela — APPROVED** [review]
Perfect!. The name of the PR better not using underscores _ but PR is short and clear and nice. Well done

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java [hunk 1]
```diff
@@ -13,8 +13,13 @@ public enum ChallengeStatus {ENDED("ENDED"),
     }
 
     public static ChallengeStatus determineChallengeStatus(String status){
```
Comment (AlexandraBonetCanela) [inline itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: I think determine is not a word commonly used in coding as it does not clearly express what the method does, resolve, parse, map or getChallengeStatusFromString may be options?

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java [hunk 2]
```diff
@@ -13,8 +13,13 @@ public enum ChallengeStatus {ENDED("ENDED"),
...
-        return status != null  && status.equalsIgnoreCase(ChallengeStatus.ENDED.getValue()) ?
-                ChallengeStatus.ENDED : null;
+        ChallengeStatus output = null;
+        if(status != null  && status.equalsIgnoreCase(ChallengeStatus.ENDED.getValue())){
```
Comment (AlexandraBonetCanela) [inline itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: No need to test status != null

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java [hunk 3]
```diff
@@ -13,8 +13,13 @@ public enum ChallengeStatus {ENDED("ENDED"),
...
+            output = ChallengeStatus.ENDED;
+        } else if (status != null && status.equalsIgnoreCase(ChallengeStatus.IN_PROGRESS.getValue())) {
```
Comment (AlexandraBonetCanela) [inline itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: No need to test status != null

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java [hunk 4]
```diff
@@ -13,8 +13,13 @@ public enum ChallengeStatus {ENDED("ENDED"),
     }
 
     public static ChallengeStatus determineChallengeStatus(String status){
-        return status != null  && status.equalsIgnoreCase(ChallengeStatus.ENDED.getValue()) ?
-                ChallengeStatus.ENDED : null;
+        ChallengeStatus output = null;
+        if(status != null  && status.equalsIgnoreCase(ChallengeStatus.ENDED.getValue())){
+            output = ChallengeStatus.ENDED;
```
Comment (AlexandraBonetCanela) [inline itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: You can make a return here inside

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java [hunk 5]
```diff
@@ -13,8 +13,13 @@ public enum ChallengeStatus {ENDED("ENDED"),
...
+        } else if (status != null && status.equalsIgnoreCase(ChallengeStatus.IN_PROGRESS.getValue())) {
+            output = ChallengeStatus.IN_PROGRESS;
```
Comment (AlexandraBonetCanela) [inline itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: Same as above, you can meka  return here

**Review by AlexandraBonetCanela — CHANGES_REQUESTED** [review]

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 1 above)
Comment (Almami679) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: The method was already created by Gwen, I just expanded it.

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 2 above)
Comment (Almami679) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: In the service, the value is managed when it is null to throw an exception, that's why I continued controlling it like this

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 3 above)
Comment (Almami679) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: In the service, the value is managed when it is null to throw an exception, that's why I continued controlling it like this

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 4 above)
Comment (Almami679) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: According to good practices, a function should only have one return.

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 2 above)
Comment (AlexandraBonetCanela) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: Oki

**Review by AlexandraBonetCanela — APPROVED** [review]

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 2 above)
Comment (ivilarop) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: Avoid repeating status != null in each condition.You can make a guard clause at the beginning

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 1 above)
Comment (ivilarop) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: Okay, Gwen did it, but if we detect that it can be improved, we can apply the change, right?

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java [hunk 6]
```diff
@@ -12,8 +13,13 @@ public enum ChallengeStatus {ENDED("ENDED");
     }
 
     public static ChallengeStatus determineChallengeStatus(String status){
-        return status != null  && status.equalsIgnoreCase(ChallengeStatus.ENDED.getValue()) ?
-                ChallengeStatus.ENDED : null;
+        ChallengeStatus output = null;
+        if(status != null  && status.equalsIgnoreCase(ChallengeStatus.ENDED.getValue())){
```
Comment (ivilarop) [inline itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: Could we simplify this by replacing the if/else with a more functional approach, such as using a Map<String, ChallengeStatus> or a stream-based solution? It might improve readability and scalability if more statuses are added in the future.

**Review by ivilarop — CHANGES_REQUESTED** [review]

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 1 above)
Comment (Almami679) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: Yes, precisely I always want to change things that I see wrong, but I focused on a few changes, I'll update it right now

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 6 above)
Comment (Almami679) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java]: I see the map more for enums with many values, and I think that a solution will not have more than 3 or 4 states, I have used stream, so that it is more readable.

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java [hunk 7]
```diff
@@ -2,18 +2,27 @@
...
     private final String value;
 
     ChallengeStatus(String value) {
         this.value = value;
     }
 
-    public static ChallengeStatus determineChallengeStatus(String status){
-        return status != null  && status.equalsIgnoreCase(ChallengeStatus.ENDED.getValue()) ?
-                ChallengeStatus.ENDED : null;
+    public static ChallengeStatus challengeStatusFromString(String status) {
```
Comment (ivilarop) [inline itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java:17]: Good use of streams! To make this even cleaner and safer, I’d suggest returning an Optional<ChallengeStatus> instead of using .orElse(null), since it brings us back to dealing with null.

Here's a proposed change:
public static Optional<ChallengeStatus> fromString(String status) {
    if (status == null) return Optional.empty();
    return Arrays.stream(ChallengeStatus.values())
        .filter(s -> status.equalsIgnoreCase(s.getValue()))
        .findFirst();
}
This way, the method clearly communicates that the result may be absent, and the caller can handle it using .orElse, .orElseThrow, .ifPresent, etc., without risking a NullPointerException.

**Review by ivilarop — CHANGES_REQUESTED** [review]

File: itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java (hunk 7 above)
Comment (Almami679) [reply itachallenge-user/src/main/java/com/itachallenge/user/document/enums/ChallengeStatus.java:17]: As we have discussed, it is left as null, since it is controlled in the service layer

**Review by ivilarop — APPROVED** [review]
